triplea-game/triplea
# Make "Flag Display Mode" menu items mutually exclusive with radio buttons.

Before this change, the items did not have decorators indicating which is selected. This change adds this functionality.
Note: In 1.9, this submenu was different and there were checkboxes next to the items, but the changes to this menu in 2.0 made the options mutually exclusive already, so this PR expands on that by making them use a proper radio UI.

## Functional Changes
[] New map or map update
[] New Feature
[X] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[] Bug fix:  
[] Other:   


## Testing

[] No manual testing done
[X] Manually tested

Testing:
1.  Open the "Flags Display Mode" submenu in the View menu on the game.
2. Verify that an item is selected and matches the display mode.
3. Toggle the state and verify that the menu items are updated with an indicator about which is active.
4. Toggle the state from the "Unit Scroller" UI and check that the selected menu item gets updated accordingly.
 
## Screens Shots

### Before
![before_radio](https://user-images.githubusercontent.com/17648/66710096-8f098880-ed3f-11e9-9c0b-40cc30978161.png)

### After

![radio](https://user-images.githubusercontent.com/17648/66710071-481b9300-ed3f-11e9-8cdd-af461db884f5.png)

Note: On Mac, radio menu items are still displayed with checkboxes. On other platforms, the indicators may be different.
